craft_application/launchpad/util.py

```diff
@@ -133,6 +133,11 @@ def get_architecture(name: str) -> Architecture:
     :raises: ValueError if there's no way to convert the string into an architecture name.
     """
     name = name.strip().lower()
+
+    # 'build-for [all]' can build anywhere, so use amd64 builders because they are fast
+    if name == "all":
```

Comment:
This should be converted elsewhere (in the calling function/method) - "all" is still not a valid `processor` in Launchpad (see `get_processor` below and https://api.launchpad.net/devel.html#processor).

Reply:
Good point.  I'm thinking about moving it to `craft_application.launchpad.models.recipe.SnapRecipe.new()`.  

I think it should live somewhere in craft-application and that seems like a reasonable place.

Reply:
I'd like to keep `craft_application.launchpad` as a pretty thin layer around launchpadlib, so putting it somewhere in the `RemoteBuildService` is probably the most sensible way to do it. (Maybe in `_new_recipe`?)

Reply:
Nice, I like that.  Moved.